core/config/toml/types.go

```diff
@@ -2163,9 +2168,44 @@ type ConnectorGateway struct {
 	URL *string
 }
 
+type SharedPeering struct {
+	Enabled       bool
+	Bootstrappers *[]ocrcommontypes.BootstrapperLocator
+	StreamConfig  StreamConfig
+}
+
+func (c *SharedPeering) setFrom(f *SharedPeering) {
+	c.Enabled = f.Enabled
+	if f.Bootstrappers != nil {
+		c.Bootstrappers = f.Bootstrappers
+	}
+	c.StreamConfig.setFrom(&f.StreamConfig)
+}
+
+type StreamConfig struct {
+	IncomingMessageBufferSize  int
+	OutgoingMessageBufferSize  int
+	MaxMessageLenBytes         int
+	MessageRateLimiterRate     float64
+	MessageRateLimiterCapacity uint32
+	BytesRateLimiterRate       float64
+	BytesRateLimiterCapacity   uint32
```

Comment:
All the fields need to be pointers in order to distinguish between unset from zero value.
```suggestion
	IncomingMessageBufferSize  *int
	OutgoingMessageBufferSize  *int
	MaxMessageLenBytes         *int
	MessageRateLimiterRate     *float64
	MessageRateLimiterCapacity *uint32
	BytesRateLimiterRate       *float64
	BytesRateLimiterCapacity   *uint32
```

Reply:
(and then `setFrom` needs to nil check each one for the layering to work properly)